test/cqlpy/test_lwt.py

```diff
@@ -103,6 +108,40 @@ def test_lwt_with_batch_conflict_2(cql, table1):
     for r in rs:
         assert r.applied == False
 
+# Moreover, there are cases where Cassandra prevents mixing
+# IF EXISTS with different conditions such as IF r=1, despite
+# the fact that the conditions are not contradictory.
+def test_lwt_with_batch_conflict_3(cql, table1, scylla_only):
+    p = unique_key_int()
+    cql.execute(f'INSERT INTO {table1}(p,c,r) VALUES ({p},1,1)')
+    rs = cql.execute(f'BEGIN BATCH UPDATE {table1} SET r=10 WHERE p={p} AND c=1 IF r=1; UPDATE {table1} SET r=20 WHERE p={p} AND c=1 IF EXISTS;  APPLY BATCH;')
+    # Cassandra fails with: "Cannot mix IF conditions and IF EXISTS for the same row"
+    for r in rs:
+        assert r.applied == True
+
+# During a static column update, the clustering key is not required in the WHERE clause.
+# A batch composed of a query that updates a static column and a query that inserts
+# a new row is allowed and will be applied successfully.
+def test_lwt_with_batch_conflict_4(cql, table1):
+    p = unique_key_int()
+    cql.execute(f'INSERT INTO {table1}(p,c,r,s) VALUES ({p},1,1,1)')
+    rs = cql.execute(f'BEGIN BATCH UPDATE {table1} SET s=NULL WHERE p={p} IF EXISTS; INSERT INTO {table1}(p,c,r) VALUES ({p},2,2) IF NOT EXISTS; APPLY BATCH;')
+    for r in rs:
+        assert r.applied == True
+
+# However, even for static columns, mixing IF EXISTS and other conditions
+# is disallowed in Cassandra.
+def test_lwt_with_batch_conflict_5(cql, table1, scylla_only):
+    p = unique_key_int()
+    cql.execute(f'INSERT INTO {table1}(p,c,r,s) VALUES ({p},1,1,1)')
+    rs = cql.execute(f'BEGIN BATCH UPDATE {table1} SET s=NULL WHERE p={p} IF EXISTS; UPDATE {table1} SET s=NULL WHERE p={p} IF s = 1; APPLY BATCH;')
+    # Cassandra fails with: "Cannot mix IF conditions and IF NOT EXISTS for the same row"
+    # The fact that "IF NOT EXISTS" is used in the error message is a bit misleading,
+    # but that's what Cassandra returns
+    for r in rs:
+        assert r.applied == True
```

Comment:
Interesting, in this, like test_lwt_with_batch_conflict_3, it's not clear to me if this is a missing Cassandra feature or an outright bug - if it's the latter we should prefer the "cassadra_bug" label over the "scylla_only" one (the difference is mostly cosmetic).

If you have the inclination to do so, you can even report these suspected bugs to the Cassandra project, and see what they think. Maybe they'll fix it, or maybe they can explain (?) how their weird behavior is actually correct or necessary for some reason.